File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/repository/CollectCategoryRepository.java
package com.windea.demo.cloudcollect.core.repository;

import com.windea.demo.cloudcollect.core.domain.entity.CollectCategory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface CollectCategoryRepository extends JpaRepository<CollectCategory, Long> {
	Optional<CollectCategory> findByName(String name);

	Page<CollectCategory> findByUser_Id(Long userId, Pageable pageable);

	long countByUser_Id(Long userId);

	Page<CollectCategory> findByUser_IdAndNameContains(Long userId, String name, Pageable pageable);

	boolean existsByUser_IdAndName(Long userId, String name);
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/repository/CollectCategoryRepository.kt
package com.windea.demo.cloudcollect.core.repository

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*
import org.springframework.data.jpa.repository.*

interface CollectCategoryRepository : JpaRepository<CollectCategory, Long> {
	fun findByNameAndUserId(name: String, userId: Long): CollectCategory?
	
	fun findAllByNameContainsAndUserId(name: String, userId: Long, pageable: Pageable): Page<CollectCategory>
	
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<CollectCategory>
	
	fun countByUserId(userId: Long): Long
	
	fun existsByNameAndUserId(name: String, userId: Long): Boolean
}
